src/Data/Foldable/Extra.hs

```diff
@@ -0,0 +1,9 @@
+module Data.Foldable.Extra where
+
+import Data.Foldable
+
+sumOn' :: (Foldable f, Num b) => (a -> b) -> f a -> b
```

Comment:
Worth saying `A generalisation of `Data.List.Extra.sumOn'`?

Reply:
Completed! Also added `Data.Foldable.Extra.sum'` and `Data.Foldable.Extra.product'` for consistency.

I'm not well-versed in INLINE pragmas; I'm not sure if adding one to any of these would have any effect.

Reply:
I tend to leave INLINE pragmas off unless someone benchmarks them as having benefit - so no need to start now.